docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-1339/README.md

```diff
@@ -0,0 +1,115 @@
+)
+```
+
+This code produces the following output accurately:
+
+```text
+5 items bought, $0.33 each. Current account balance: $1.35
+```
+
+## Compliant Solution (Decimal)
+
+This compliant solution adheres to standards by utilizing the imported `Decimal` type, which allows for precise representation of decimal values. It's important to note that, on most platforms, calculations using `Decimal` tend to be less efficient compared to those using basic data types.
+
+*[compliant02.py](compliant02.py):*
+
+```py
+
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
+""" Compliant Code Example """
+from decimal import Decimal
+BALANCE = Decimal("3.00")
+ITEM_COST = Decimal("0.33")
+ITEM_COUNT = 5
+ 
+print(
+    f"{str(ITEM_COUNT)} items bought, ${ITEM_COST} each. "
+    f"Current account balance: "
+    F"${str(BALANCE - ITEM_COUNT * ITEM_COST)}"
+)
+```
+
+This code outputs the following:
+
+```text
+5 items bought, $0.33 each. Current account balance: $1.35
+```
+
+## Automated Detection
+
+|Tool|Version|Checker|Description|
+|:----|:----|:----|:----|
+|[Pylint](https://pylint.pycqa.org/)|2023.10.1|Not Available|Not detected|
+
+## Related Guidelines
+
+|||
+|:---|:---|
+|[SEI CERT Oracle Coding Standard for Java](https://wiki.sei.cmu.edu/confluence/display/java/SEI+CERT+Oracle+Coding+Standard+for+Java?src=breadcrumbs)|[NUM04-J. Do not use floating-point numbers if precise computation is required](https://wiki.sei.cmu.edu/confluence/display/java/NUM04-J.+Do+not+use+floating-point+numbers+if+precise+computation+is+required)|
+|[The CERT C Secure Coding Standard](https://wiki.sei.cmu.edu/confluence/display/c/SEI+CERT+C+Coding+Standard)|[FLP02-C. Avoid using floating-point numbers when precise computation is needed](https://wiki.sei.cmu.edu/confluence/display/c/FLP02-C.+Avoid+using+floating-point+numbers+when+precise+computation+is+needed)
+|[The CERT C++ Secure Coding Standard](https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=88046682) |[VOID FLP02-CPP. Avoid using floating point numbers when precise computation is needed](https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=88046687)|
+|[ISO/IEC TR 24772:2010](http://www.aitcnet.org/isai/)| Floating-Point Arithmetic [PLF]|
+|[MITRE CWE Pillar]| [CWE-682: Incorrect Calculation](https://cwe.mitre.org/data/definitions/682.html)|
+|[MITRE CWE Base]|[CWE - CWE-1339: Insufficient Precision or Accuracy of a Real Number](https://cwe.mitre.org/data/definitions/1339.html)|
+
+## Bibliography
+
+|||
+|:---|:---|
+|[Bloch 2008](https://wiki.sei.cmu.edu/confluence/display/java/Rule+AA.+References#RuleAA.References-Bloch08)|Item 48, "Avoid `float` and `double` If Exact Answers Are Required"|
+|[Bloch 2005](https://wiki.sei.cmu.edu/confluence/display/java/Rule+AA.+References#RuleAA.References-Bloch05)|Puzzle 2, "Time for a Change"|
+|[IEEE 754](https://wiki.sei.cmu.edu/confluence/display/java/Rule+AA.+References#RuleAA.References-IEEE7542006)||

```

Comment:
the links are links to links, since Bloch is a book we can't link to it just reference it. Not sure if we can link to the IEEE

```suggestion
|[Bloch 2008]|Item 48, "Avoid `float` and `double` If Exact Answers Are Required"|
|[Bloch 2005]|Puzzle 2, "Time for a Change"|
|[IEEE 754]||
```